src/presentation/api/controller/marketplace.go

```diff
@@ -41,49 +40,104 @@ func (controller *MarketplaceController) ReadCatalog(c echo.Context) error {
 	)
 }
 
-func (controller *MarketplaceController) parseDataFields(
-	rawDataFields []interface{},
-) ([]valueObject.MarketplaceInstallableItemDataField, error) {
-	dataFields := []valueObject.MarketplaceInstallableItemDataField{}
-
-	for fieldIndex, rawDataField := range rawDataFields {
-		errPrefix := "[index " + strconv.Itoa(fieldIndex) + "] "
+func (controller *MarketplaceController) transformDataFieldsIntoMap(
+	rawDataFields string,
+) []map[string]interface{} {
+	dataFieldsMapSlice := []map[string]interface{}{}
+	if len(rawDataFields) == 0 {
+		return dataFieldsMapSlice
+	}
 
-		rawDataFieldMap, assertOk := rawDataField.(map[string]interface{})
-		if !assertOk {
-			return dataFields, errors.New(errPrefix + "InvalidDataFieldStructure")
+	rawDataFieldsSlice := strings.Split(rawDataFields, ";")
+	for _, rawDataField := range rawDataFieldsSlice {
+		rawDataFieldParts := strings.Split(rawDataField, ":")
+		if len(rawDataFieldParts) != 2 {
+			slog.Error(
+				"InvalidDataFieldStringStructure",
+				slog.String("rawDataField", rawDataField),
+			)
+			continue
 		}
 
-		rawName, exists := rawDataFieldMap["name"]
-		if !exists {
-			rawName, exists = rawDataFieldMap["key"]
-			if !exists {
-				return dataFields, errors.New(errPrefix + "DataFieldNameNotFound")
-			}
-			rawDataFieldMap["name"] = rawName
-		}
+		dataFieldsMapSlice = append(
+			dataFieldsMapSlice,
+			map[string]interface{}{rawDataFieldParts[0]: rawDataFieldParts[1]},
+		)
+	}
 
-		fieldName, err := valueObject.NewDataFieldName(rawName)
+	return dataFieldsMapSlice
+}
+
+func (controller *MarketplaceController) parseDataFieldMap(
+	rawDataFields map[string]interface{},
+) []valueObject.MarketplaceInstallableItemDataField {
+	dataFields := []valueObject.MarketplaceInstallableItemDataField{}
+
+	for rawFieldName, rawFieldValue := range rawDataFields {
+		fieldName, err := valueObject.NewDataFieldName(rawFieldName)
 		if err != nil {
-			return dataFields, errors.New(errPrefix + err.Error())
+			slog.Error(err.Error(), slog.String("rawFieldName", rawFieldName))
+			continue
 		}
 
-		fieldValue, err := valueObject.NewDataFieldValue(rawDataFieldMap["value"])
+		fieldValue, err := valueObject.NewDataFieldValue(rawFieldValue)
 		if err != nil {
-			return dataFields, errors.New(errPrefix + err.Error())
+			slog.Error(err.Error(), slog.Any("rawFieldValue", rawFieldValue))
+			continue
```

Comment:
That's a great insight!
I hadn't thought of it from that perspective! I'll pay more attention to it next time.

Reply:
That's a great insight! I hadn't considered it from that perspective! I'll keep it in mind for next time.

As for the change, to add the index, I'll need to initialize an artificial one outside the for loop since I'm iterating over a map. I don't think this will be an issue given the context in which this index will be used and the self-explanatory name.